tsparser/src/parser/resources/infra/objects.rs

```diff
@@ -123,11 +144,106 @@ pub fn resolve_bucket_usage(data: &ResolveUsageData, bucket: Lrc<Bucket>) -> Res
     })
 }
 
+fn parse_bucket_ref(
+    data: &ResolveUsageData,
+    bucket: Lrc<Bucket>,
+    _call: &MethodCall,
+    type_arg: &ast::TsType,
+) -> Result<Option<Usage>> {
+    fn process_type(
+        data: &ResolveUsageData,
+        sp: &swc_common::Span,
+        t: &Type,
+        depth: usize,
+    ) -> Option<Vec<Operation>> {
+        if depth > 10 {
+            // Prevent infinite recursion.
+            return None;
+        }
+
+        match t {
+            Type::Named(named) => {
+                let ops = match named.obj.name.as_deref() {
+                    Some("Lister") => vec![Operation::ListObjects],
+                    Some("Exister") | Some("Attrser") => vec![Operation::GetObjectMetadata],
```

Comment:
The `Exister` permissions here isn't referenced in the docs and in the go-code, is it in use somewhere else?

Reply:
Good catch, fixed :)